syntax = "proto3";
import "google/protobuf/empty.proto";

message Genres {
    int32 id = 1;
    string name = 2;
}

message Movies {
    bool adult = 1;
    string language = 2;
    string title = 3;
    repeated Genres genres = 4;
}

message Data {
    repeated Movies movies = 1;
}

message MessageResponse {
    bool success = 1;
}

message MoviesData {
    bool adult = 1;
    string language = 2;
    string title = 3;
}

message MoviesList {
    repeated MoviesData movies = 1;
}

message MovieId {
    int32 movie_id = 1;
}


service MoviesService {
    rpc GetMovies(google.protobuf.Empty) returns (MoviesList) {}
    rpc GetMovie(MovieId) returns (Movies) {}
    rpc UpdateMovies (Data) returns (MessageResponse) {}
}